### PR TITLE
L2 lambda for Odin API

### LINKED ISSUE #2: Script for adding tasks
Some sort of script for adding tasks is needed. Look at https://github.com/Odin-SMR/qqjobs for inspiration, but dumb down if possible! Also look at how it is currently executed in production, so see what the actual requirements are.

``` cron
###
## qqjobs
###
15 06 * * * docker pull -q odinsmr/microq_admin > /dev/null
45 06 * * * docker run --rm -v ~/.odin.cfg:/odin.cfg:ro odinsmr/microq_admin add-production-jobs
```

### PR DESCRIPTION
**Pull request by skymandr**
I think this should work for the processing system.

Two stacks:
* Job stack creates lambda which regularly (once per day) tries to add new jobs to queues
    * This stack also creates all queues for all projects and frequency mode
* Worker stack creates one lambda per project and frequency mode coupled with a docker image for that combination
    * The worker stacks wake up on new messages in the queues and tries to process, posting results to API
    * DLQ collects failed jobs

Resolves #2

### CODE REVIEW

File: stacks/level2_stack.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+FROM odinsmr/u-jobs:qsmr_{invmode}_{freqmode}_{project}
+
+FROM public.ecr.aws/lambda/python:3.9
+
+# Copy function code
+COPY ./level2/handlers/level2.py /var/task
+
+# Set the CMD to your handler
+CMD [ "level2.lambda_handler" ]
+"""
```
Comment (skymandr) [inline stacks/level2_stack.py:24]: Is this all it takes, do you think?

File: stacks/level2_stack.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    DockerImageFunction,
+)
+from aws_cdk.aws_lambda_event_sources import SqsEventSource
+from aws_cdk.aws_sqs import DeadLetterQueue, Queue
+from constructs import Construct
+
+DOCKER_IMAGE = """
+FROM odinsmr/u-jobs:qsmr_{invmode}_{freqmode}_{project}
+
+FROM public.ecr.aws/lambda/python:3.9
```
Comment (joakimmoller) [inline stacks/level2_stack.py:17]: This looks a bit strange, this looks like a multistage build but no artefact is brought from the first image to the second. So I guess you will not have anything to work with

usually something is taken from the first image like this:
```
FROM odinsmr/u-jobs:qsmr_{invmode}_{freqmode}_{project} as builder
RUN something
FROM public.ecr.aws/lambda/python:3.9
COPY --from=builder /datafile /var/data/datafile
...
```
or in this case when we just want to pick some files (many?)
```
FROM public.ecr.aws/lambda/python:3.9
COPY --from=odinsmr/u-jobs:qsmr_{invmode}_{freqmode}_{project} /datafile /var/data/datafile
...
```

Comment (skymandr) [reply stacks/level2_stack.py]: Typ så? https://github.com/Odin-SMR/odin-l2-lambda/pull/1/commits/04d9acaaa71595ea424060991b826c32b29eafa0

File: stacks/level2_jobs_stack.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        lambda_schedule: Schedule = Schedule.rate(Duration.days(1)),
+        queue_retention_period: Duration = Duration.days(14),
+        **kwargs
+    ) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        lambda_name = "OdinSMRLevel2JobsLambda"
+        processing_table_name = "OdinSMRLevel2ProcessingTable"
+
+        level2_jobs_lambda = PythonFunction(
```
Comment (joakimmoller) [inline stacks/level2_jobs_stack.py:36]: I think you can replace to "Fuction" as it is. the runtime arg tells its python.

**Review by joakimmoller — APPROVED** [review]
Let's se if it works

File: stacks/level2_jobs_stack.py (hunk 3 above)
Comment (skymandr) [reply stacks/level2_jobs_stack.py:36]: Detta blir skillnaden:

``` python
-        level2_jobs_lambda = PythonFunction(
+        level2_jobs_lambda = Function(
             self,
             lambda_name,
-            entry="level2",
-            handler="handler",
-            index="handlers/level2_jobs.py",
+            code=InlineCode.from_asset("../level2/handlers"),
+            handler="level2_jobs.handler",
             runtime=Runtime.PYTHON_3_9,
             timeout=lambda_timeout,
             environment={
```

**Pull request merged**